packages/cube/lib/cube/buildCubeShape/Constraints.js

```diff
@@ -0,0 +1,210 @@
+export class CompositeConstraintBuilder {
+  constructor(...builders) {
+    this.builders = builders
+  }
+
+  add(object) {
+    this.builders.forEach(builder => builder.add(object))
+  }
+
+  build(ptr) {
+    this.builders.forEach(builder => builder.build(ptr))
+  }
+}
```

Comment:
Surprised that eslint doesn't complain about multiple classes in one module. I think it's a good rule. 

Right now, for example, it was not obvious for me that there are multiple functionalities provided by the `Constraints.js` module.

I would propose to move each to a subdir, like `constrainBuilder/CompositeConstraintBuilder.js` etc and leave `Constraints.js` re-exporting them if needed. I would have the test file also split to avoid a very large test suite which I find hard to navigate

Reply:
Independent testability is one of the advantages of the approach proposed here, so splitting the test file definitively makes sense to me. But for the implementation classes I feel like breaking the 'one file per class' rule (as long as the classes are small and cohesive) because I tend to miss the big picture with too many classes scattered around. I know in enterprisey OO this rule is a must (and since I brought up OO patterns here I have to accept it :)).